src/clj/rems/db/core.clj

```diff
@@ -2,17 +2,27 @@
   {:ns-tracker/resource-deps ["sql/queries.sql"]}
   (:require [clj-time.core :as time]
             [clj-time.jdbc] ;; convert db timestamps to joda-time objects
+            [clojure.java.jdbc]
             [clojure.set :refer [superset?]]
             [conman.core :as conman]
             [mount.core :refer [defstate] :as mount]
             [rems.config :refer [env]]))
 
 (defstate ^:dynamic *db*
-  :start (cond
-           (:test (mount/args)) (conman/connect! {:jdbc-url (:test-database-url env)})
-           (:database-url env) (conman/connect! {:jdbc-url (:database-url env)})
-           (:database-jndi-name env) {:name (:database-jndi-name env)}
-           :else (throw (IllegalArgumentException. ":database-url or :database-jndi-name must be configured")))
+  :start (let [db (cond (:test (mount/args)) (conman/connect! {:jdbc-url (:test-database-url env)})
+                        (:database-url env) (conman/connect! {:jdbc-url (:database-url env)})
+                        (:database-jndi-name env) {:name (:database-jndi-name env)}
+                        :else (throw (IllegalArgumentException. ":database-url or :database-jndi-name must be configured")))]
+           (try
+             (with-open [_ (clojure.java.jdbc/get-connection db)]
+               nil)
+             (catch Exception e
+               (throw (IllegalArgumentException.
+                       (str "Can not connect to database "
+                            (pr-str db)
+                            ". :database-name or :database-jndi-name invalid.")
```

Comment:
I think typically this would be `Check xxx or yyy`. There could be some other connection problem in theory?

Reply:
reworded